classes/user/Collector.inc.php

```diff
@@ -330,254 +342,260 @@ public function offset(?int $offset): self
+        // Break words by whitespace, trims and escapes "%" and "_"
+        $words = array_map(fn(string $word) => '%' . addcslashes($word, '%_') . '%', PKPString::regexp_split('/\s+/', $searchPhrase));
+        foreach ($words as $word) {
+            $query->where(
+                fn() => $query->whereRaw('LOWER(u.username) LIKE LOWER(?)', [$word])
+                    ->orWhereRaw('LOWER(u.email) LIKE LOWER(?)', [$word])
+                    ->orWhereExists(
+                        fn(Builder $subQuery) => $subQuery->from('user_settings', 'us')
                             ->whereColumn('us.user_id', '=', 'u.user_id')
-                            ->where('us.setting_name', '=', $settingMap[$this->orderBy])
-                            ->whereIn('us.locale', $locales)
-                            ->orderByRaw("COALESCE(us.setting_value, '') = ''") // Not empty values first
-                            ->orderByRaw('CASE us.locale ' . str_repeat('WHEN ? THEN ?', count($locales)) . ' ELSE ? END', $bindinds) // Then follow the locale order
-                            ->limit(1)
-                            ->select('us.setting_value');
-                    }, $this->orderDirection);
-                },
-                function (Builder $query): void {
-                    $sortMap = [self::ORDERBY_ID => 'u.user_id'];
-                    if (!isset($sortMap[$this->orderBy])) {
-                        throw new InvalidArgumentException("Invalid order by: {$this->orderBy}");
-                    }
-                    $query->orderBy($sortMap[$this->orderBy], $this->orderDirection);
-                }
+                            ->whereIn('us.setting_name', $settings)
+                            ->whereRaw('LOWER(us.setting_value) LIKE LOWER(?)', [$word])
+                    )
+                    ->orWhereExists(
+                        fn(Builder $subQuery) => $subQuery->from('user_interests', 'ui')
+                            ->join('controlled_vocab_entry_settings AS cves', 'ui.controlled_vocab_entry_id', '=', 'cves.controlled_vocab_entry_id')
+                            ->whereColumn('ui.user_id', '=', 'u.user_id')
+                            ->whereRaw('LOWER(cves.setting_value) LIKE LOWER(?)', [$word])
+                    )
             );
+        }
 
-        // Add app-specific query statements
-        HookRegistry::call('User::Collector', [$q, $this]);
+        return $this;
+    }
 
-        return $q;
+    /**
+     * Handles the order by clause
+     */
+    protected function buildOrderBy(Builder $query): self
+    {
+        $orderByFields = [self::ORDERBY_ID => 'u.user_id'];
+        if ($orderByField = $orderByFields[$this->orderBy] ?? null) {
+            $query->orderBy($orderByField, $this->orderDirection);
+            return $this;
+        }
+
+        $settings = [self::ORDERBY_GIVENNAME => Identity::IDENTITY_SETTING_GIVENNAME, self::ORDERBY_FAMILYNAME => Identity::IDENTITY_SETTING_FAMILYNAME];
```

Comment:
When sorting by given name, isn't it interesting to add the last name to the sorting (and vice-versa)?

Reply:
Sure, if it's easy to add.

Reply:
@asmecher Ok, I've updated.